x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/NativeMemoryCapacity.java

```diff
@@ -7,39 +7,60 @@
 
 package org.elasticsearch.xpack.ml.autoscaling;
 
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import org.elasticsearch.common.unit.ByteSizeValue;
 import org.elasticsearch.xpack.autoscaling.capacity.AutoscalingCapacity;
 import org.elasticsearch.xpack.ml.utils.NativeMemoryCalculator;
 
 import java.util.Objects;
-import java.util.Optional;
 
-import static org.elasticsearch.xpack.ml.utils.NativeMemoryCalculator.dynamicallyCalculateJvmSizeFromNativeMemorySize;
+import static org.elasticsearch.xpack.ml.MachineLearning.NATIVE_EXECUTABLE_CODE_OVERHEAD;
 
-// Used for storing native memory capacity and then transforming it into an autoscaling capacity
-// which takes into account the whole node size
+/**
+ * Used for storing native memory capacity and then transforming it into an autoscaling capacity
+ * which takes into account the whole node size.
+ */
 public class NativeMemoryCapacity {
 
     static final NativeMemoryCapacity ZERO = new NativeMemoryCapacity(0L, 0L);
 
+    private static final Logger logger = LogManager.getLogger(NativeMemoryCapacity.class);
+
     static NativeMemoryCapacity from(NativeMemoryCapacity capacity) {
-        return new NativeMemoryCapacity(capacity.tierMlNativeMemoryRequirement, capacity.nodeMlNativeMemoryRequirement, capacity.jvmSize);
+        return new NativeMemoryCapacity(
+            capacity.tierMlNativeMemoryRequirementExcludingOverhead,
+            capacity.nodeMlNativeMemoryRequirementExcludingOverhead,
+            capacity.jvmSize
+        );
     }
 
-    private final long tierMlNativeMemoryRequirement;
-    private final long nodeMlNativeMemoryRequirement;
+    private final long tierMlNativeMemoryRequirementExcludingOverhead;
+    private final long nodeMlNativeMemoryRequirementExcludingOverhead;
```

Comment:
This change to doing capacity calculations in terms of memory _excluding_ the per-node overhead considerably simplifies the calculations.  It allows other code to avoid the need to think about how many nodes will be added on a scale-up.